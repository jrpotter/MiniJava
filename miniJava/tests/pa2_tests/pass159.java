// PA1 parse decl pass
class Test {

    int [] a;
    Test [] t;

    void p() {
        void x = y;
    }
}

