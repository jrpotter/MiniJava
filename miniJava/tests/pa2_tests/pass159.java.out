======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "Test" classname
  .   FieldDeclList [2]
  .   . (public) FieldDecl
  .   .   ArrayType
  .   .     INT BaseType
  .   .   "a" fieldname
  .   . (public) FieldDecl
  .   .   ArrayType
  .   .     ClassType
  .   .       "Test" classname
  .   .   "t" fieldname
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     VOID BaseType
  .   .   .     "x" varname
  .   .   .   RefExpr
  .   .   .     IdRef
  .   .   .       "y" Identifier
=============================================
